scripts/zkai_native_attention_mlp_adapter_air_frontier_gate.py

```
@@ -0,0 +1,646 @@
+            source_artifact("native_attention_mlp_single_proof_gate", SINGLE_PROOF_PATH, single, single_raw),
+            source_artifact("native_attention_mlp_lifting_ablation_gate", LIFTING_ABLATION_PATH, lifting, lifting_raw),
+        ],
+    }
+
+
+def validate_expected_sources(sources: dict[str, Any]) -> None:
+    derived_summary = _dict(sources["derived"].get("summary"), "derived input summary")
+    for key, expected in EXPECTED_DERIVED_INPUT.items():
+        actual = sources["derived"].get(key) if key in {"schema", "decision", "result"} else derived_summary.get(key)
+        if actual != expected:
+            raise AdapterAirFrontierError(f"derived input {key} drift")
```

Ruff (0.15.12)

[warning] Avoid specifying long messages outside the exception class

(TRY003)